src/main/java/io/github/opencubicchunks/cubicchunks/core/CommonEventHandler.java

```diff
@@ -176,4 +179,23 @@ private boolean shouldSkipWorld(World world) {
         return !allowedServerWorldClasses.contains(world.getClass())
                 || !allowedServerChunkProviderClasses.contains(world.getChunkProvider().getClass());
     }
+
+    @SubscribeEvent
+    public void onWorldUnload(WorldEvent.Unload event) {
+        if (event.getWorld().isRemote || !((ICubicWorld) event.getWorld()).isCubicWorld()) {
+            return;
+        }
+
+        ICubicWorld world = (ICubicWorld) event.getWorld();
+        if (!world.isCubicWorld()) {
+            return;
+        }
+
+        ICubeIO io = ((ICubeProviderInternal.Server) world.getCubeCache()).getCubeIO();
+        try {
+            io.close();
```

Comment:
I would expect this to break things

Reply:
the event is fired immediately before the vanilla code flushes the write queue - (but using a different flush method that never gets forwarded along to `ICubeIO`). i'm *almost* certain that nothing will ever be written afterwards, and if it does, `AsyncBatchingCubeIO` will throw an exception which can be debugged.

Reply:
I have seen "already closed" exceptions even in the previous code being thrown (only sometimes, rarely) while quitting the world or shutting down MC, which is why I'm suspicious of just closing it.

Reply:
from what i can tell, unless there's another mod doing something totally wrong, this should be totally safe. all chunks will have been unloaded before the event is fired, the only stuff that happens afterwards should be `WorldServer#flush()` -> `AnvilSaveHandler#flush()` (which does nothing in cubic worlds), followed by `DimensionManager` marking the world as unloaded.